tests/integration/test_dynamodb.py

```diff
@@ -560,6 +560,35 @@ def test_dynamodb_stream_records_with_update_item(self):
 
         dynamodb.delete_table(TableName=table_name)
 
+    def test_query_on_deleted_resource(self):
+        table_name = 'ddb-table-%s' % short_uid()
+        partition_key = 'username'
+
+        dynamodb = aws_stack.connect_to_service('dynamodb')
+        aws_stack.create_dynamodb_table(table_name, partition_key)
+
+        rs = dynamodb.query(
```

Comment:
nitpick: We could add a small util function for the query, to avoid duplication with lines 582-588.